repositories/go_repositories.bzl

```diff
@@ -0,0 +1,46 @@
+# Copyright 2016 The Bazel Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Once recursive workspace is implemented in Bazel, this file should cease
+# to exist.
+"""
+Provides functions to pull all Go external package dependencies of this
+repository.
+"""
+
+load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")
+
+def go_deps():
+    """Pull in external Go packages needed by Go binaries in this repo.
+
+    Pull in all dependencies needed to build the Go binaries in this
```

Comment:
Why not just add these to repositories.bzl?

Reply:
This file has
`load("@bazel_gazelle//:deps.bzl", "gazelle_dependencies", "go_repository")`
at the top which won't work in repositories.bzl because the macro `repositories` imports the bazel_gazelle repository. Thus, this *has* to be a separate file.

Reply:
would it be possible then to call this from repositories.bzl?